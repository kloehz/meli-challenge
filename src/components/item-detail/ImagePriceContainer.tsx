import useCurrency from "../../hooks/useCurrency";
import { Item } from "../../types/getItemDetail"
import { Prices } from "../prices/Prices";


export const ImagePriceContainer = ({picture, condition, sold_quantity, title, price}: Item) => {
    const parsedPrice = useCurrency({...price});
    return (
        <div className="item-image-price-container">
            <div className="item-detail-image-container">
                <img src={picture}/>
            </div>
            <div className="item-detail-to-buy-container">
                <span className="item-detail-condition-selled" >
                    {condition} - {sold_quantity} vendido(s)
                </span>
                <h1 className="item-detail-title">
                    {title}
                </h1>
                <Prices price={parsedPrice} style='item-detail' />
            <button className="item-detail-buy-button">Comprar</button>
            </div>
        </div>
    )
}
