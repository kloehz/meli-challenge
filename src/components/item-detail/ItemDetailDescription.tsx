interface IDesc {
    description: string
}

export const ItemDetailDescription = ({description}: IDesc) => {
    return (
        <div className="item-detail-description-container">
            <span className="desc-title">
                Descripcion del producto
            </span>
            <span className="desc-info">
                {description}
            </span>
        </div>
    )
}
