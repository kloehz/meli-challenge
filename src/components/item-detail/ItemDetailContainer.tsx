import { Item } from "../../types/getItemDetail"
import { Seo } from "../seo/Seo";
import { ImagePriceContainer } from "./ImagePriceContainer";
import { ItemDetailDescription } from "./ItemDetailDescription";

export const ItemDetailContainer = (props: Item) => {
    return (
            <>
                <Seo title={props.title}/>
                <div className="item-detail-container">
                    <ImagePriceContainer {...props} />
                    <ItemDetailDescription  description={props.description} />
                </div>
            </>
    )
}
