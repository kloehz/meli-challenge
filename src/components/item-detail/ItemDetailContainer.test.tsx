import { shallow } from "enzyme"
import { ImagePriceContainer } from "./ImagePriceContainer";
import { ItemDetailContainer } from "./ItemDetailContainer"
import { ItemDetailDescription } from "./ItemDetailDescription";

const itemMock = {
    "id":"MLA912120163",
        "title":"Set X6 Vasos Vidrio Color Trago Largo 465 Cc Pulsar Bormioli",
        "price":{"currency":"ARS",
        "amount":1499,
        "decimals":0},
        "picture":"http://http2.mlstatic.com/D_625911-MLA45228911173_032021-O.jpg",
        "condition":"Nuevo",
        "free_shipping":false,
        "sold_quantity":50,
        "description":"descripcion"
    }

describe('ItemDetailContainer tests', () => {
    
    const wrapper = shallow(<ItemDetailContainer {...itemMock}/>);

    test('should be match snapshoot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should be render whith the correct description', () => {
        expect(wrapper.containsMatchingElement(<ItemDetailDescription description={'descripcion'}/>))
        .toEqual(true);
    });

    test('should be render whith the correct props', () => {
        expect(wrapper.containsMatchingElement(<ImagePriceContainer {...itemMock}/>))
        .toEqual(true);
    })
    
    
    
})
