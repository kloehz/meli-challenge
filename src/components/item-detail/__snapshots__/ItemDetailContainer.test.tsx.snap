// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemDetailContainer tests should be match snapshoot 1`] = `ShallowWrapper {}`;
